cli/src/cmd/forge/verify.rs

```diff
@@ -93,20 +99,35 @@ pub async fn run_verify(args: &VerifyArgs) -> eyre::Result<()> {
     };
 
     let project = build_args.project()?;
-    let contract = project
-        .flatten(&project.root().join(args.contract.path.as_ref().unwrap()))
-        .map_err(|err| eyre::eyre!("Failed to flatten contract: {}", err))?;
+
+    let (source, contract_name) = if args.flatten {
+        // NOTE: user need to set bytecodehash='ipfs' for this otherwise verification won't work
+        // see: https://github.com/gakonst/foundry/issues/1236
```

Comment:
I still don't understand how this is a verification issue when the bytecodehash does not play a role here?

I'm referring to the API request not what etherscan sets when it tries to verify.

Reply:
@mattsse when we submit verification using flattened file format, etherscan will compile the contract using `solc` default metadata bytecode hash (`ipfs`).

The metadata bytecode hash will be appended to the bytecode, hence the bytecode will missmatch (foundry compile the contract with `bytecodehash=none` while etherscan compile it with `bytecodehash=ipfs`)

Reply:
```
// Metadata settings (optional)
    "metadata": {
      // Use the given hash method for the metadata hash that is appended to the bytecode.
      // The metadata hash can be removed from the bytecode via option "none".
      // The other options are "ipfs" and "bzzr1".
      // If the option is omitted, "ipfs" is used by default.
      "bytecodeHash": "ipfs"
    },
```

@mattsse more info https://docs.soliditylang.org/en/v0.8.13/using-the-compiler.html#compiler-input-and-output-json-description

Reply:
yes that's perfectly clear, but I'd rather consider `bytecodehash=none` a deployment "issue"/choice

anyway, thanks for this!

Reply:
> but I'd rather consider bytecodehash=none a deployment "issue"/choice

@mattsse yes mee too, it just how etherscan works by default 

> anyway, thanks for this!

U r welcome ser 👍